### PR TITLE
Edwin - Added user classes to weekly summary report

### PR DESCRIPTION
**Pull request by EdwinLauM**
# Description

Screen Shot 2023-07-25 at 12 04 08 PM

## Related PRS (if any):

This frontend PR is related to the BE PR [PR #458](https://github.com/OneCommunityGlobal/HGNRest/pull/458)

## Main changes explained:

- Made code changes to show the user class in the weekly summary report
- Added check to not display 'Volunteer' role
- Made slight changes to backend for weekly summary request to now include the users role

## How to test:

1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as any user that has the view weekly summary permission (Owner/Admin)
4. Dashboard -> Reports -> Weekly Summaries Report
5. Check that the user class for every user is shown on the right of the google docs icon
6. Users whose class is Volunteer, should not show since that is the default 

## Videos/Screenshots of changes:

Screen Shot 2023-07-25 at 12 08 47 PM

### CODE REVIEW

**Review by sxiong5 — APPROVED** [review]
Hey, the feature worked as expected! Great job on the code style.

Screenshot 2023-07-25 at 15 52 11

**Review by BaileyM96 — APPROVED** [review]
Hello @EdwinLauM I reviewed your PR and followed your directions, and everything works and looks great, great work!

## Screenshots

PR #1042 user classes to weekly summary report

**Review by veronicacheng2 — APPROVED** [review]
Hi @EdwinLauM, I tested the PR and everything works as expected

Role Listing for roles other than Volunteer:
image

**Review by winghojackyli — APPROVED** [review]
Hi @EdwinLauM , I tested your PR and it works great. Using admin account, I can see the roles of users near the google icon except for volunteer.

**Comment by beblicarl** [comment]
This functionality works as expected but the dev environment lags way too much. I am sure it's a general problem

issue #1042

**Comment by Shreyj14** [comment]
Hi @EdwinLauM everything worked great!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118795427/020d90f2-4653-42b0-9f6b-d2cdd0cabbaa

**Review by Shreyj14 — APPROVED** [review]

**Comment by iTvX** [comment]
I have tested your PR, and everything looks good. Great job!
CleanShot 2023-07-26 at 12 25 49@2x

**Review by eduvarjaor — APPROVED** [review]
This feature works as intended. Great job!
Screenshot_1

**Review by igorrochadasilva — APPROVED** [review]
Hi @EdwinLauM !
I hope you are well =D 
I tested your PR and everything works correcty, congratulations and great job!
1042_01_weekly-summary-report class showed

**Review by DavidC0126 — APPROVED** [review]
Hi @EdwinLauM ,
I tested your PR and it works as intended. Nice job!
user_classes

**Comment by mounica-dingari-by** [comment]
Hi @EdwinLauM , I've tested this PR and it works as expected. Great Job!

Screenshot from 2023-07-26 23-18-54

**Review by xaanders — APPROVED** [review]
Hi @EdwinLauM ! I've tested it and it works good! All the classes are shown. Good job.

**Comment by AnishPandita** [comment]
Hi @EdwinLauM, tested the feature and it works well.
Screenshot 2023-07-27 at 3 13 35 PM

**Review by vemanamadhu — APPROVED** [review]
Everything looks good and works as expected. Good job.
PR_1042

I had one question though:
1. Why isn't there a separate field called role. Wouldnt that be more readable? Because when I first opened the page I had to look for some time for changes that were made.

Let me know what you think @EdwinLauM and @one-community

**Review by yaow62023 — COMMENTED** [review]
I tested your PR, and most of the features work fine. However, there is one issue where the position is not being displayed within the parentheses. Strangely, some reviews show the role inside the parentheses, while others do not. Check the screen shoot I provided. 

Screenshot 2023-07-28 at 11 46 29 PM

File: src/components/WeeklySummariesReport/FormattedReport.jsx [hunk 1]
```diff
@@ -228,6 +228,9 @@ const FormattedReport = ({ summaries, weekIndex, bioCanEdit }) => {
               <span onClick={() => handleGoogleDocClick(googleDocLink)}>
                 <img className="google-doc-icon" src={google_doc_icon} alt="google_doc" />
               </span>
+              <span>
+                <b>&nbsp;&nbsp;{summary.role !== 'Volunteer' && `(${summary.role})`}</b>
```
Comment (iTvX) [inline src/components/WeeklySummariesReport/FormattedReport.jsx:232]: Maybe use CSS will better than using &nbsp;&nbsp. For example, classname the code google, and write the css below:
`.google {
  margin-left: 10px;
}`

**Review by mahamakhan — APPROVED** [review]
Works as mentioned.
image

**Comment by zuhang2000** [comment]
Hi Edwin @EdwinLauM ! I tested your PR following the instructions on both backend and frontend. Everything works as expected! I am able to see different roles beside people's names.
Screen Shot 2023-07-28 at 3 06 00 PM

**Review by zuhang2000 — APPROVED** [review]
Everything work as expected! I test both the front-end and back-end. The roles are shown properly.
Screen Shot 2023-07-28 at 3 06 00 PM

**Review by shiva2096 — APPROVED** [review]
Hi @EdwinLauM 

Great work! Functionality is working as expected.

Screenshot 2023-07-28 at 8 18 19 PM

**Review by ramyaram2092 — CHANGES_REQUESTED** [review]
Hello @EdwinLauM  i see  some glitch . Its unable to fetch the  class name . Could you take a look at this 
image

**Comment by Lucas-E** [comment]
Thank you everyone! Moving this one to final review!

**Comment by EdwinLauM** [comment]
> Hello @EdwinLauM i see some glitch . Its unable to fetch the class name . Could you take a look at this image

Hey @ramyaram2092, are you running this frontend with its respective backend PR?

**Pull request merged**